src/utils.ts

```diff
@@ -0,0 +1,68 @@
+import { ContributionPartyEvent } from "types";
+import { ethersProvider } from "./ethereum";
+import {
+  haveSetNewPartyWithContribution,
+  haveSetPartyHalfway,
+  setNewPartyWithContribution,
+  setPartyHalfway,
+} from "./party_events";
+
+export const bestUserName = async (address: string): Promise<string> => {
```

Comment:
wonder if we should look up OS username if no ens is set?

Reply:
i like that. cool if i save that for the next release?

Reply:
yesser